web/api/v0/__init__.py

```diff
@@ -343,6 +346,97 @@ def get(self, ipv4: IPv4Address | IPv6Address) -> ResponseReturnValue:
 
 api.add_resource(UserByIPResource, '/user/from-ip')
 
+class MPSKSClientAddResource(Resource):
+    @use_kwargs(
+        {
+            "password": fields.Str(required=True),
+            "mac": fields.Str(required=True),
+            "name": fields.Str(required=True),
+        },
+        location="form",
+    )
+    def post(self, user_id: int, password: str, mac: str, name: str) -> ResponseReturnValue:
+        user = get_authenticated_user(user_id, password)
+
+        try:
+            # checks rather the user has all settable mpsks clients created
+            if len(user.mpsks) > current_app.config.get("MAX_MPSKS", 30):
+                abort(400, message="User has the maximum count of mpsk clients.")
+
+            if not user.wifi_password:
+                abort(400, message="Legacy wifi password change of password is required.")
+
+            mpsk_client_create(session.session, user, mac, name, user)
```

Comment:
@agmes4 BTW, there was a subtle bug here: name and mac are flipped.
For this reason, it is really helpful to always use keyword arguments, even if it seems redundant: `name=name, mac=mac, …`

Im thinking about whether it makes sense to start enforcing [keyword-only arguments](https://docs.python.org/3/tutorial/controlflow.html#keyword-only-arguments).

(I'm on fixing this myself, so click `resolve` if you've read and understood this)